modyn/config/examples/modyn_config.yaml

```diff
@@ -67,7 +70,7 @@ storage:
         selector_batch_size: 256,
       },
       {
-        name: "yearbook-test",
+        name: "yearbook_test",
```

Comment:
@robinholzi let's try to make as few breaking changes as possible until the paper submission, because on the sigmod branch there are scripts with hard-coded dataset names. Thanks! 

The scripts won't run properly without a name change. CC @MaxiBoether 

PS: not related to this PR, just want to bring it to attention that we shouldn't change the default value in Pydantic configs in following PRs until submission, because otherwise the scripts on sigmod branch will run suddently with different configs without notice.